### PR TITLE
:bug: upgrade golangci-lint to v1.61.0

### PR DESCRIPTION
**Pull request by sivchari**
Upgade the golangci-lint version to v1.61.0.

### CODE REVIEW

File: pkg/cache/cache_test.go [hunk 1]
```diff
@@ -1544,7 +1544,7 @@ func CacheTest(createCacheFunc func(config *rest.Config, opts cache.Options) (ca
 					return obtainedPodNames
 				}, ConsistOf(tc.expectedPods)))
 				for _, pod := range obtainedStructuredPodList.Items {
-					Expect(informer.Get(context.Background(), client.ObjectKeyFromObject(&pod), &pod)).To(Succeed()) //nolint:gosec // We don't retain the pointer
+					Expect(informer.Get(context.Background(), client.ObjectKeyFromObject(&pod), &pod)).To(Succeed()) // We don't retain the pointer
```
Comment (alvaroaleman) [inline pkg/cache/cache_test.go:1547]: Can delete the entire comments for these, the purpose of the comment was to explain the `nolint` directive

File: .golangci.yml [hunk 2]
```diff
@@ -162,6 +162,9 @@ issues:
     - linters:
         - gosec
       text: "G304: Potential file inclusion via variable"
+    - linters:
+        - gosec
+      text: "G115: integer overflow conversion"
```
Comment (sbueringer) [inline .golangci.yml:167]: I would prefer not globally ignoring this finding. Let's plese revert the change here and take a look at the occurences, verify that it's fine to ignore there and then add //nolint comments if appropriate

Comment (sbueringer) [reply .golangci.yml:167]: I think the one in admission is fine, the one in zap/flags.go could probably use some additional validation

File: pkg/log/zap/flags.go [hunk 3]
```diff
@@ -85,7 +85,7 @@ func (ev *levelFlag) Set(flagValue string) error {
 		}
 		if logLevel > 0 {
 			intLevel := -1 * logLevel
-			ev.setFunc(zap.NewAtomicLevelAt(zapcore.Level(int8(intLevel))))
+			ev.setFunc(zap.NewAtomicLevelAt(zapcore.Level(int8(intLevel)))) //nolint:gosec // We don't care about G115.
```
Comment (sbueringer) [inline pkg/log/zap/flags.go:88]:
```suggestion
			ev.setFunc(zap.NewAtomicLevelAt(zapcore.Level(int8(intLevel)))) //nolint:gosec // We are not worried about integer overflows (G115) here .
```

File: pkg/webhook/admission/response.go [hunk 4]
```diff
@@ -71,7 +71,7 @@ func ValidationResponse(allowed bool, message string) Response {
 		AdmissionResponse: admissionv1.AdmissionResponse{
 			Allowed: allowed,
 			Result: &metav1.Status{
-				Code:   int32(code),
+				Code:   int32(code), //nolint:gosec // We don't care about G115.
```
Comment (sbueringer) [inline pkg/webhook/admission/response.go:74]:
```suggestion
				Code:   int32(code), //nolint:gosec // Integer overflows (G115) cannot occur here .
```

File: tools/setup-envtest/store/store.go [hunk 5]
```diff
@@ -167,8 +167,8 @@ func (s *Store) Add(ctx context.Context, item Item, contents io.Reader) (resErr
 		// preferfing our own scheme.
 		targetPath := filepath.Base(header.Name)
 		log.V(1).Info("writing archive file to disk", "archive file", header.Name, "on-disk file", targetPath)
-		perms := 0555 & header.Mode // make sure we're at most r+x
-		binOut, err := itemPath.OpenFile(targetPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, os.FileMode(perms))
+		perms := 0555 & header.Mode                                                                        // make sure we're at most r+x
+		binOut, err := itemPath.OpenFile(targetPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, os.FileMode(perms)) //nolint:gosec // We don't care about G115.
```
Comment (sbueringer) [inline tools/setup-envtest/store/store.go:171]:
```suggestion
		binOut, err := itemPath.OpenFile(targetPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, os.FileMode(perms)) //nolint:gosec // Integer overflows (G115) seem unlikely here.
```

**Comment by sbueringer** [comment]
@sivchari Do you have time to address the findings? I might need this PR so I can bump to Kubernetes v1.32 alpha.1

**Review by vincepri — APPROVED** [review]

**Comment by sivchari** [comment]
@sbueringer 
cc. @vincepri 
Thanks, I didn't have time to work on it, so I appreciate it so much.

**Comment by sbueringer** [comment]
No worries!
